src/lib.zig

```diff
@@ -25,3 +29,16 @@ pub fn trace(comptime src: std.builtin.SourceLocation) callconv(.Inline) Ctx {
         .c = c.___tracy_emit_zone_begin_callstack(&loc, 1, 1),
     };
 }
+
+pub fn traceName(comptime src: std.builtin.SourceLocation, comptime name: [*c]const u8) callconv(.Inline) Ctx {
```

Comment:
since this was more of an oversight on my part, can you make `name` optional and then make `traceName` the new `trace`?

Reply:
Sure!  One question: Do you mean as an [optional type](https://ziglang.org/documentation/master/#Optionals) or does Zig support some kind of optional parameters?

Reply:
optional parameters is something i hope makes it eventually into zig but yes i was referring to `?`

Reply:
also it should be `[]const u8` and if the struct needs `[*c]const u8` we should then pass it `name.ptr`

Reply:
Also done.  I am not quite happy with the `if (name) |*n| n.ptr else null` but I don't see a nicer/simpler way to do it.

Reply:
hmm you're prolly right, sorry about that. sticking with `?[*c]const u8` would probably be best instead of erasing the type

Reply:
But are we erasing the type there?  I thought we just provide a nice default with slightly uncomfortable syntax.

I think `?[]const u8` with the if statement hidden inside is fine, but I can change it back to `?[*c]const u8` if you like!

Reply:
so sorry for the delay, yeah lets go with `?[*c]const u8` and then this looks good to go

Reply:
also the `name` in `trace` doesn't need to be marked as comptime

Reply:
Okay, finally done with this, please merge if it looks all right!  Also thanks for getting through the review with me, I know it's a been back-and-forth. 😅